lib/results_validators/advancement_conditions_validator.rb

```diff
@@ -94,11 +94,11 @@ def run_validation(validator_data)
 
                 # Check that no one proceeded if they shouldn't have
                 if condition.instance_of? AdvancementConditions::AttemptResultCondition
-                  current_persons = results_by_round_type_id[round_type_id].map(&:wca_id)
+                  current_persons = results_by_round_type_id[round_type_id].map(&:person_id)
                   people_over_condition = previous_results.filter do |r|
                     sort_by_column = r.format.sort_by == "single" ? :best : :average
                     current_persons.include?(r.wca_id) && r.send(sort_by_column) > condition.attempt_result
```

Comment:
If `current_persons` now consists of lots of `person_id`s, then you should not look up the `wca_id` here. How the heck did the tests pass on this?

Reply:
Oh wow, good catch. It passes because `current_persons.include(r.wca_id)` is always false, and my new test only checks that this error _doesn't_ get raised. I thought the other tests covered this, but apparently not. This is why you _always_ include a failure case when testing, lol